Port to ROS 2 (foxy).

pgold opened the pull request:
CMake required version bumped to 3.5.
package.xml version bumped to 3.

pgold committed:
Port to ROS 2 (foxy).

CMake required version bumped to 3.5.
package.xml version bumped to 3.

jacobperron commented on README.md:
```diff
@@ -63,21 +63,20 @@ _* Not verified. Anyone who is able to verify that this driver works or not is e
 * Internet connection
-* [ROS](http://wiki.ros.org/ROS/Installation) _Indigo_ or _Jade_
-* Ubuntu packages: `python-rosdep`, `python-catkin-tools`
+* [ROS](http://wiki.ros.org/ROS/Installation) _Foxy_
```
This installation link should point to https://index.ros.org/doc/ros2/Installation/#installationguide

pgold replied:
Fixed.

jacobperron commented on README.md:
```diff
@@ -63,21 +63,20 @@ _* Not verified. Anyone who is able to verify that this driver works or not is e
-* [ROS](http://wiki.ros.org/ROS/Installation) _Indigo_ or _Jade_
-* Ubuntu packages: `python-rosdep`, `python-catkin-tools`
+* [ROS](http://wiki.ros.org/ROS/Installation) _Foxy_
+* Ubuntu packages: `python3-rosdep2`, `python3-colcon-common-extensions`
```
I think `python3-rosdep` should continue to work. Is there an issue?

pgold replied:
No issue. I followed Ubuntu's suggestion of installing `python3-rosdep2`, but from reading the docs I think `python3-rosdep` is indeed the correct one.

jacobperron commented on README.md:
````diff
@@ -124,17 +123,17 @@ $ sudo apt-get install python-rosdep python-catkin-tools
 For Create 2 (Roomba 600/700 series):
 ``` bash
-$ roslaunch create_bringup create_2.launch
+$ ros2 launch create_bringup create_2.xml
````
I don't think renaming the launch files is necessary. Correct me if I'm wrong.

pgold replied:
I tried to use `.launch`, but `ros2 launch` would try to interpret them as python ROS2 launch files. Adding the `.xml` made it correctly parse then as XML launch files, but there might be a better way of doing it -- I'm open to suggestions.

jacobperron commented on create_bringup/CMakeLists.txt:
```diff
@@ -1,11 +1,13 @@
 project(create_bringup)
 
-find_package(catkin REQUIRED)
+set(CMAKE_CXX_STANDARD 14)
```
There's no C++ code in this package. I guess we can remove this line.

pgold replied:
Removed.

jacobperron commented on create_bringup/package.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<package format="2">
+<package format="3">
```
Why format 3? I think we can continue to use format 2.

pgold replied:
So that all package.xml files would be on the same version. Reverted.

jacobperron commented on create_description/CMakeLists.txt:
```diff
@@ -1,25 +1,23 @@
-)
-
-catkin_package()
+set(CMAKE_CXX_STANDARD 14)
```
I think we can remove this line.

pgold replied:
Done.

jacobperron commented on create_description/package.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<package format="2">
+<package format="3">
```
I think we can keep this format 2.

pgold replied:
Done.

jacobperron commented on create_driver/package.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<package format="2">
+<package format="3">
```
Keep format 2

pgold replied:
Done.

jacobperron commented on create_msgs/CMakeLists.txt:
```diff
@@ -1,25 +1,24 @@
-  message_generation
-  std_msgs
-)
+set(CMAKE_CXX_STANDARD 14)
```
Is this necessary?

pgold replied:
Not for `create_msgs`.

jacobperron commented on create_robot/package.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<package format="2">
+<package format="3">
```
Keep format 2

pgold replied:
Done.

jacobperron commented on create_driver/src/create_driver.cpp:
```diff
@@ -31,22 +31,31 @@ POSSIBILITY OF SUCH DAMAGE.
     nh_(nh),
-    priv_nh_("~"),
-    diagnostics_(),
+    priv_nh_(rclcpp::Node::make_shared("_")),
```
Since parameters are "private" to a node in ROS 2, I don't think we need to a separate node anymore. Please remove `priv_nh_` and just use `nh_` for declaring parameters below.

pgold replied:
Done.

jacobperron commented on create_driver/src/create_driver.cpp:
```diff
@@ -31,22 +31,31 @@ POSSIBILITY OF SUCH DAMAGE.
+  priv_nh_->declare_parameter<double>("loop_hz", 10.0);
+  priv_nh_->declare_parameter<bool>("publish_tf", true);
+
+  priv_nh_->get_parameter("dev", dev_);
```
nitpick: we can effectively get the value from the return of `declare_parameter` above. For example:

```c
dev_ = nh_->declare_parameter<std::string>("dev", "/dev/ttyUSB0");
```

This will make the code a bit more compact.

pgold replied:
Good catch -- I'm not very familiar with this interface. Done.

jacobperron reviewed (commented):
Thanks a lot for the port!

It generally looks good. Left some minor comments below.

pgold committed:
Fix README prerequisites.

pgold committed:
Remove unneeded CMAKE_CXX_STANDARD.

pgold committed:
Revert package.xml's to version 2.

As these don't need to be on version 3.

pgold committed:
Remove CreateDriver's priv_nh_ member.

pgold committed:
Use value returned by Node::declare_parameter.

pgold committed:
Remove uneeded changes.

create_description's CMakeLists and package.xml was missing on the
previous commits.

jacobperron commented on create_driver/include/create_driver/create_driver.h:
```diff
@@ -64,56 +64,56 @@ class CreateDriver
-  ros::Publisher wheeldrop_pub_;
-  ros::Publisher wheel_joint_pub_;
+  std::shared_ptr<rclcpp::Node> nh_;
+  std::shared_ptr<rclcpp::Node> priv_nh_;
```
We can remove this extra node reference now.

pgold replied:
Good catch. Done.

jacobperron commented on README.md:
````diff
@@ -124,17 +123,17 @@ $ sudo apt-get install python-rosdep python-catkin-tools
 For Create 2 (Roomba 600/700 series):
 ``` bash
-$ roslaunch create_bringup create_2.launch
+$ ros2 launch create_bringup create_2.xml
````
I think we should rename the launch files so they have the extension `*.launch.xml`. This seems to make tab-completion work when executing `ros2 launch create_bringup <tab>`.

pgold replied:
Please disregard my previous comment regarding launch files. I've renamed them to `.launch` and it's working as expected.

jacobperron reviewed (commented):
Two more comments to address, then I think it is good to go.

pgold committed:
Remove left-over member from CreateDriver.

pgold committed:
Rename launch files.

From *.xml to *.launch.

jacobperron reviewed (approved):
Cheers!
